### PR TITLE
feat(filtered-search): support free text as pills

### PR DESCRIPTION
**Pull request by spike-rabbit**
This adds a new experimental feature, which will allow adding free text as pills. Free-text are handled as normal Criteria, except that they are defined in a separate input. We need a separate input due to their different behavior.

Every new part of the API is flagged as experimental, since future changes are expected.

Documenation is lacking on purpose. It will be added in a follow-up.
See the playground for usage.

> Describe in detail what your merge request does and why. Add relevant
> screenshots and reference related issues via `Closes #XY` or `Related to #XY`.

---

- [x] I confirm that this MR follows the [contribution guidelines](https://github.com/siemens/element/blob/main/CONTRIBUTING.md).

## Summary

Adds experimental support for free-text pills to the filtered-search component, allowing arbitrary text entries to be created and edited as criteria.

### Key changes
- Free-text UI & behavior
  - New inputs freeTextCriterion and searchForFreeTextLabel; new SiFilteredSearchFreeTextComponent for editing.
  - createFreeTextPill(query) and freeTextBlur() integrated into typeahead/template to create free-text pills.
  - Playground: "Disable free text pills" toggle and conditional freeTextCriterion binding.
- Interceptor & permissions
  - DisplayedCriteriaEventArgs.allow signature extended to allow(criteriaNamesToDisplay, freeText?: boolean).
  - allowFreeTextCache added and interceptor integration included (was planned separately but merged into this PR).
- API & translations
  - InternalCriterionDefinition accepts optional type?: 'free-text'.
  - New translatable key SI_FILTERED_SEARCH.SEARCH_FOR_FREE_TEXT.
  - New API surface marked experimental; documentation deferred to follow-up.
- Tests & tooling
  - New unit tests, an e2e test for free-text flows, multiple Playwright snapshots, and harness filter extension (valueText).

### Known issues / review notes
- strictCriterion validation currently flags free-text criteria invalid; planned fix via a future validation function.
- Semicolon (`;`) handling deferred to a separate PR.
- UX: pressing the search button while typing free text does not auto-create a pill.
- Edge case: entering `abc:` in the free-text input can remove the pill unexpectedly.

### CODE REVIEW

File: playwright/e2e/element-examples/si-filtered-search.spec.ts [hunk 1]
```diff
@@ -79,4 +79,33 @@ test.describe('filtered search', () => {
...
+    await expect(page.getByLabel('search', { exact: true }).first()).toBeFocused();
+    await si.runVisualAndA11yTests('freetext-edit-mode');
+    // Clear and type new value
+    await page.keyboard.press('Control+KeyA');
+    await page.keyboard.type('updated free text');
+    await page.keyboard.press('Enter');
+    // Verify the value was updated
+    await expect(page.getByText('updated free text', { exact: true })).toBeVisible();
+    await si.runVisualAndA11yTests('freetext-updated');
+  });
```
[the remark this answers is not in the document]
Comment (spike-rabbit) [reply playwright/e2e/element-examples/si-filtered-search.spec.ts]: - Direct search button submission is covered in another text
- semicolon is moved to anohter PR
- strictCriterion is also moved to another PR

File: projects/element-ng/filtered-search/si-filtered-search.component.ts [hunk 2]
```diff
@@ -709,4 +749,30 @@ export class SiFilteredSearchComponent implements OnInit, OnChanges {
...
+    const freeTextDef = this.freeTextCriterion();
+    if (!freeTextDef) {
+      return;
+    }
+    const internalDef: InternalCriterionDefinition = {
+      ...freeTextDef,
+      label: freeTextDef.label ?? freeTextDef.name,
+      translatedLabel: freeTextDef.label ?? freeTextDef.name,
+      type: 'free-text'
+    };
```
[the remark this answers is not in the document]
Comment (spike-rabbit) [reply projects/element-ng/filtered-search/si-filtered-search.component.ts]: yes, this is on prupose, as the label is unsed, by our types cannot reflect this currently. Will be fixed in a follow-up PR.

File: projects/element-ng/filtered-search/values/free-text/si-filtered-search-free-text.component.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    class="px-4 py-0 border-0 focus-inside"
+    [attr.aria-label]="searchLabel() | translate"
+    [disabled]="disabled()"
+    [value]="criterionValue().value"
+    (input)="freeTextValueChange($event)"
+    (keydown.backspace)="valueBackspace()"
+    (keydown.enter)="valueEnter()"
+    (blur)="active.set(false)"
+  />
+}
```
[the remark this answers is not in the document]
Comment (spike-rabbit) [reply projects/element-ng/filtered-search/values/free-text/si-filtered-search-free-text.component.html]: this is covered by already exising tests.

**Comment by spike-rabbit** [comment]
As discussed offline with @spliffone :
- `strictCriterion` handling won't be adjusted. Instead we will introduce a new validation function to also solve many other short comings in that area
- handling of `;` is also moved to another PR

File: projects/element-ng/filtered-search/testing/si-filtered-search-criterion.harness.ts [hunk 4]
```diff
@@ -59,7 +60,15 @@ export class SiFilteredSearchCriterionHarness extends ComponentHarness {
...
-      );
+      )
+      .addOption('valueText', filters.valueText, async (harness, text) => {
+        const valueElement = await harness.value();
+        if (!valueElement) {
+          return false;
+        }
+        const value = await valueElement.text();
+        return HarnessPredicate.stringMatches(Promise.resolve(value), text);
+      });
```
[the remark this answers is not in the document]
Comment (spike-rabbit) [reply projects/element-ng/filtered-search/testing/si-filtered-search-criterion.harness.ts]: We have prettier, so no

**Review by kfenner — APPROVED** [review]
LGTM 👍
